bento_wes/backends/wes_backend.py

```diff
@@ -252,6 +257,119 @@ def get_workflow_name_wdl(workflow_path: Path) -> str | None:
             # Invalid/non-workflow-specifying WDL file if false-y
             return workflow_id_match.group(1) if workflow_id_match else None
 
+    def _download_to_path(self, url: str, token: str, destination: str):
+        """
+        Download a file from a URL to a destination directory.
+        Bearer token auth works with Drop-Box and DRS.
+        """
+        with requests.get(
+            url,
+            headers={"Authorization": f"Bearer {token}"},
+            verify=current_app.config["BENTO_VALIDATE_SSL"],
+            stream=True
+        ) as response:
+            if response.status_code != 200:
+                raise RunExceptionWithFailState(
+                    STATE_EXECUTOR_ERROR,
+                    f"Download request to drop-box resulted in a non 200 status code: {response.status_code}"
+                )
+            with open(destination, 'wb') as f:
+                # chunk_size=None to use the chunk size from the stream
+                for chunk in response.iter_content(chunk_size=None):
+                    f.write(chunk)
+        self.log_debug(f"Downloaded file at {url} to path {destination}")
+
+    def _inputs_url_refs(self, files: str | list[str]) -> str | list[str] | None:
+        if not files:
+            return None
+
+        if isinstance(files, list):
+            input_files = [get_drop_box_resource_url(f_path) for f_path in files]
+        else:
+            input_files = get_drop_box_resource_url(files)
+        return input_files
+
+    def _download_input_files(self, inputs: str | list[str], token: str, run_dir: Path) -> str | list[str] | None:
+        if not inputs:
+            return None
```

Comment:
does it matter that this returns `None` instead of `[]` if an empty list is passed in as an input?